src/main/java/ui/components/NavigableListView.java

```diff
@@ -216,4 +222,11 @@ public void selectNextItem() {
         return selectedIndex.map(getItems()::get);
     }
 
+    /**
+     * @return The integer value of the selected index in the list panel or -1 if it is null.
+     */
+    public int getSelectedIndex() {
+        return selectedIndex.isPresent() ? selectedIndex.get() : -1;
+    }
```

Comment:
This method is basically replaces null with -1 and then you check for -1 later. Why not just return the `selectedIndex` instead and let the caller deals with null?